overlord/devicestate/devicestate.go

```diff
@@ -1114,6 +1102,29 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 	return tss, nil
 }
 
+func ensureModelBaseIsGadgetBase(new *asserts.Model, tracker *snap.SelfContainedSetPrereqTracker) error {
```

Comment:
Nitpicking with names here: maybe `checkRequiredGadgetMatchesModelBase` and `s/new/model/` for the variable name.